### DISCUSSION TITLE
Update to a new point cloud as a widget

### DISCUSSION THREAD

**Discussion by julienguegan** [pyvista/pyvista#2270] [Q&A]
Hi guys !

I am new to pyVista and I wanted to know if it was possible to update a RGB point cloud that I have in my plotter (with `pl.add_mesh(point_cloud, rgb=True))` by another one ? I have seen that it is possible to `update_scalars` and `update_coordinates`, also there is overwrite but I am not sure that is the solution ... Ideally , I would like to have the same plotter directly updated without having to call `pl.show()` in another cell (like a ipywidget). The new point cloud is a different one that I read with `point_cloud = pv.read(filename)`

Thanks ! :slightly_smiling_face:

Note that it seems that it can be done with pythreejs https://github.com/jupyter-widgets/tutorial/blob/master/notebooks/extras/pythreejs.ipynb

**Comment by banesullivan** [comment]
You can use the `name` parameter in `add_mesh` to replace the old mesh with the new mesh. For example:

```py
import pyvista as pv
import numpy as np

sphere = pv.Sphere()

plotter = pv.Plotter(notebook=True)
plotter.add_mesh(sphere, name='sphere')
viewer = plotter.show(jupyter_backend='ipyvtklink', return_viewer=True)
viewer
```

```py
import time
for center in np.random.randn(10, 3)*.5:
    plotter.add_mesh(pv.Sphere(center=center), name='sphere')
    time.sleep(0.25)
    viewer.update_canvas()
```

https://user-images.githubusercontent.com/22067021/176056525-67e9123c-9a60-4155-bc9d-461ffca1aa80.mov
